### ISSUE TITLE
keras implementation addition

### ISSUE THREAD

**Issue by ghost** [andreyzharkov/keras-monotonic-attention#3]
Hello @andreyzharkov  can you include a sample implementation just like this one.

https://github.com/datalogue/keras-attention/issues/22

Also in the read me it is mentioned that some of these are optional

Luong-style monotonic attention 
attention weight regularization 

How to include them in the architecture

**Comment by reyzharkov** [comment]
The simple example can be found in https://github.com/andreyzharkov/keras-monotonic-attention/blob/master/example.py

As for attention type - there is a description in AttentionDecoder docstring
AttentionDecoder has params `normalize_energy` and `is_monotonic`
From the docstring:
```
        references:
            (1) Bahdanau, Dzmitry, Kyunghyun Cho, and Yoshua Bengio.
            "Neural machine translation by jointly learning to align and translate."
            arXiv preprint arXiv:1409.0473 (2014).
            (2) Colin Raffel, Minh-Thang Luong, Peter J. Liu, Ron J. Weiss, Douglass Eck
            "Online and Linear-Time Attention by Enforcing Monotonic Alignments"
            arXiv arXiv:1704.00784 (2017)
        notes:
            with `is_monotonic=True`, `normalize_energy=True` equal to model in (2)
            with `is_monotonic=False`, `normalize_energy=False` equal to model in (1)
```

**Comment by reyzharkov** [comment]
I admit that it is not absolutely obvious now where to find the example and how to use provided attention layer, so I will update the Readme file a bit later

**Comment by dcabz** [comment]
Just add a argument parser for better understanding !!

**Comment by dcabz** [comment]
Please Add  a argument parser for the same.

Also the parameters of position_embedding.py are a bit ambiguos.Please provide what each of max_time, n_waves, d_mode mean

Thanks in advance.

**Comment by dcabz** [comment]
Also the parameters of position_embedding.py are a bit ambiguos.what do each of max_time, n_waves, d_mode mean

Please Add  a argument parser for the same.

Thanks in advance.

**Comment by ghost** [comment]
Just asked for a sequential model implementation and the comments are obviously self explanatory,however could you provide a detailed explanation of what each of the variables do at different layers in example.py.

Add  a argument parser for the same for better understanding.

**Comment by ghost** [comment]
@andreyzharkov Could you do it a bit sooner,I have some deadline approaching and I need to process a output  vocabulary in the ranges of 30,000.I hope your version of code will tackle the problem.

Sorry for the annoying !

**Comment by reyzharkov** [comment]
@dcabz if you want an example similar to the one given [here](https://github.com/datalogue/keras-attention/issues/22) there it is
```python
model = Sequential()
model.add(Embedding(input_vocab.size(), input_embedding_dim))
model.add(Bidirectional(LSTM(150, return_sequences=True)))
model.add(AttentionDecoder(150, output_vocab.size(), embedding_dim=output_embedding_dim))
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['acc'])
model.summary()
```
It should deal with the problem of large vocabulary size.
Note: in that example input and output sequence lengths are equal in each batch but may be different between batches

Sorry, but I can't help you anything else until next weekend

**Comment by ghost** [comment]
Will try this out and let you know by the next weekend how your version is working.
Thanking you in advance

**Comment by ghost** [comment]
@andreyzharkov  Can you update the readme now.It is not working at all and my model is something like this having to deal with variable length sequences with large vocabularies by adding *YOUR ATTENTION MECHANISM*.It would be great if you can help with the issue

    model = keras.Sequential()
    model.add(keras.layers.Embedding(src_vocab, n_units, input_length=src_timesteps, mask_zero=True))
    model.add(LSTM(n_units))
    model.add(RepeatVector(tar_timesteps))
    model.add(LSTM(n_units, return_sequences=True))
    model.add(TimeDistributed(Dense(tar_vocab, activation='softmax')))
    return model

**Comment by reyzharkov** [comment]
@dcabz, ok, did it. And what do you mean by "it is not working at all"? Can you provide more details? See sequential_example.py it should be similar to your task. And actually the code there is almost the same as my previous reply

I will close issue for now, but feel free to reopen it if the troubles remain

**Comment by ghost** [comment]
By adding your proposition of embedding_dim=output_embedding_dim, it giving some strange errors.Anyways,I shifted to OpenNMT for using attention.Thanks

**Comment by reyzharkov** [comment]
@dcabz I cannot help when I know just that it raises "some strange errors". Be more concrete if you really want help

**Comment by ghost** [comment]
Sorry for being ambiguous ,but i am getting a error of something like embedding_dim for the output layer is not in keras.my problem is solved.

Thanks for your patience
